pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -413,7 +413,7 @@ func (l *LibvirtDomainManager) generateSomeCloudInitISO(vmi *v1.VirtualMachineIn
 		if size != 0 {
 			err = cloudinit.GenerateEmptyIso(vmi.Name, vmi.Namespace, cloudInitDataStore, size)
 		} else {
-			err = cloudinit.GenerateLocalData(vmi.Name, vmi.Namespace, cloudInitDataStore)
+			err = cloudinit.GenerateLocalData(vmi.Name, vmi.Namespace, vmi.Annotations["Flavor"], cloudInitDataStore)
```

Comment:
still references the "flavor" annotation directly instead of the ones in types.go.

To validate this is all working e2e, you might want to create a functional test that verifies the cloud-init metadata instance type is as expected within the guest.